### ISSUE TITLE
Keyboard Trap exists on New Tools button inside of Chat

### ISSUE THREAD

**Issue by accesswatch** [microsoft/vscode#262332]
Type: <b>Bug</b>

1. Open VS Code.
2. Press ctrl+alt+i.
3. Tab a few times until you hear a button talking about new tools available. At this but that is all I can tab to and pressing keys such as F6 doesn’t
seem to resolve things.
4. I must close chat and re-open it.

VS Code version: Code 1.103.1 (360a4e4fd251bfce169a4ddf857c7d25d1ad40da, 2025-08-12T16:25:40.542Z)
OS version: Windows_NT x64 10.0.26100
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|13th Gen Intel(R) Core(TM) i5-1340P (16 x 2189)|
|GPU Status|2d_canvas: enabled<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>trees_in_viz: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|15.69GB (2.01GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (77)</summary>

Extension|Author (truncated)|Version
---|---|---
jelly-language-support|Act|1.2.2
Bookmarks|ale|13.5.0
mcp-store|Aut|1.0.0
vscode-intelephense-client|bme|1.14.4
vscode-innosetup|cho|1.9.3
vscode-pandoc|Chr|0.6.2
promptboost|chr|0.8.0
npm-intellisense|chr|1.4.5
github-copilot-code-review|Cod|0.0.4
vscode-markdownlint|Dav|0.60.0
vscode-eslint|dba|3.0.16
vscode-faker|dee|2.0.0
docker|doc|0.15.0
githistory|don|0.6.20
xml|Dot|2.5.1
prettier-vscode|esb|11.0.0
vscode-firefox-debug|fir|2.15.0
macros|ged|1.2.1
codespaces|Git|1.17.3
copilot|Git|1.350.0
copilot-chat|Git|0.30.1
remotehub|Git|0.64.0
vscode-github-actions|git|0.27.2
vscode-pull-request-github|Git|0.116.1
vscode-settings-cycler|hoo|1.0.1
vscode-htmlhint|HTM|1.11.1
copy-markdown-as-html|jer|1.1.0
mark-sharp|jon|1.8.3
language-matlab|Mat|1.3.4
markdown-shortcuts|mdi|0.12.0
rainbow-csv|mec|3.20.0
copilot-code-review|mka|1.0.6
vscode-azure-mcp-server|ms-|0.5.6
vscode-containers|ms-|2.1.0
vscode-docker|ms-|2.0.0
csdevkit|ms-|1.41.11
csharp|ms-|2.87.31
vscode-dotnet-runtime|ms-|2.3.7
vscode-edge-devtools|ms-|2.1.9
datawrangler|ms-|1.22.0
remote-containers|ms-|0.425.0
remote-ssh|ms-|0.121.2025081515
remote-ssh-edit|ms-|0.87.0
remote-wsl|ms-|0.99.0
vscode-remote-extensionpack|ms-|0.26.0
live-server|ms-|0.4.15
remote-explorer|ms-|0.6.2025081809
remote-repositories|ms-|0.43.2025081801
remote-server|ms-|1.6.2025081809
vscode-speech|ms-|0.16.0
vscode-typescript-next|ms-|6.0.20250818
vscode-websearchforcopilot|ms-|0.1.2
vsliveshare|ms-|1.0.5959
windows-ai-studio|ms-|0.18.3
html-snippets-pro|rit|1.0.1
file-bunny|rob|2.4.0
markdown-shortcuts|rob|0.7.1
markdown-snippets|rob|0.9.1
marky-dynamic|rob|0.4.2
marky-edit|rob|0.7.0
marky-markdown|rob|2.10.0
marky-stats|rob|0.11.0
snippets-ranger|rob|1.8.0
joplin-vscode-plugin|rxl|1.7.3
partial-diff|ryu|1.4.3
markdown-preview-enhanced|shd|0.8.19
code-spell-checker|str|4.0.47
wav-preview|suk|2.6.0
markdown-live-preview|Tec|1.5.0
intellicode-api-usage-examples|Vis|0.2.9
vscodeintellicode|Vis|1.3.2
gitdoc|vsl|0.2.3
wordpress-toolbox|wor|1.3.16
fake-data|wyc|1.0.0
php-debug|xde|1.37.0
local-history|xyz|1.8.1
markdown-all-in-one|yzh|3.6.3

</details>

**Comment by accesswatch** [comment]
@meganrogge  We are seeing more reports of this by multiple users of this keyboard trap ...

**Comment by justschen** [comment]
which button are you getting stuck in? is it the tools button, or another one in the chat input?

**Comment by accesswatch** [comment]
@justschen  The weird issue with this bug is it tends to come and go. There used to be a button in the tab order called New Tools. I sometimes see it and other times do not which is strange but it is this button. I am having trouble getting it to appear at the moment and now wondering if a setting I have made has caued it to disappear. I tried disabling virtual tooling/tools but still no go.

**Comment by accesswatch** [comment]
@justschen  Do you by chance require more details on this issue?

**Comment by accesswatch** [comment]
@justschen  The Enterprise Disability Answer Desk has also notified the VSCode team and was able to replicate the issue.

**Comment by meganrogge** [comment]
can you pls run the command `Developer: Toggle Screencast Mode`, start a gif recording, and reproduce the issue? then attach the gif here

**Comment by accesswatch** [comment]
Here is a video (unable to attach) showing the problem @meganrogge  @justschen .  https://www.dropbox.com/scl/fi/axyxvoouip5gq8zpld26d/GMT20250823-185945_Recording_2256x1504.mp4?rlkey=vex6e1bvoa7vu8a1ixwmy499c&dl=0

**Comment by meganrogge** [comment]
Thanks so much. I can repro.

**Comment by meganrogge** [comment]
The tab trap occurs on `li.action-item-menu-entry.chat-mcp.chat-mcp-has-action`

**Comment by meganrogge** [comment]
Interesting, now it's working for me 🤔

**Comment by meganrogge** [comment]
@accesswatch can you pls try Insider's?

**Comment by accesswatch** [comment]
@meganrogge  I believe that ws the build I used when verifying the problem. It comes and goes, sometimes it works and other times it does not but many in the blind/low vision community are seeing the issue.

**Comment by meganrogge** [comment]
There's also no focused border around the item when it's focused via keyboard.

**Comment by meganrogge** [comment]
I have a repro: 
- tab to the item and wait until the hover shows
- try to tab or shift tab out of it
- 🐛 focus is trapped

It seems like this is related to the hover widget

**Comment by zuhairmahd** [comment]
I can also note similar behavior if you get an error loading an MCP tool. The keyboard is stuck on the error.
